tests/database/test_leveldb_db_backend.py

```diff
@@ -37,6 +46,136 @@ def test_set_and_get(memory_db, level_db):
     assert level_db.get(b'1') == memory_db.get(b'1')
 
 
+def test_atomic_batch(level_db, atomic_db):
```

Comment:
Rather than duplicating these here, do you think it makes sense to make this test suite generic or parametrized so that we can confirm feature parity between our in-memory databases and the leveldb ones?